pkg/api/types.go

```diff
@@ -1205,6 +1221,13 @@ type KeyToPath struct {
 	Mode *int32
 }
 
+// Local represents a directory with specific topology contraints
+type LocalVolumeSource struct {
+	// The full path to the volume on the node
+	// For alpha, this path must be a file-based volume
+	Path string
```

Comment:
For backwards compatibility I don't think we cannot require it.  I don't think it can be defaulted either.  Something like nfs is only file-based, and gce pd is only block-based.  So if it's left out, then we could resort to old behavior that it's up to the volume plugin to do the formatting.